Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -26,70 +26,85 @@ namespace Content.Shared.Damage.Components
-
-        [ViewVariables] public IReadOnlyDictionary<DamageType, int> DamageTypes => _damageList;
-
-        [ViewVariables] public HashSet<DamageType> SupportedTypes { get; } = new();
-
-        [ViewVariables] public HashSet<DamageClass> SupportedClasses { get; } = new();
-
-        public bool SupportsDamageClass(DamageClass @class)
-        {
-            return SupportedClasses.Contains(@class);
-        }
+        public HashSet<DamageGroupPrototype> SupportedGroups { get; } = new();
 
-        public bool SupportsDamageType(DamageType type)
-        {
-            return SupportedTypes.Contains(type);
-        }
+        public HashSet<DamageTypePrototype> SupportedTypes { get; } = new();
 
         protected override void Initialize()
         {
             base.Initialize();
-
-            var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+            _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
 
             // TODO DAMAGE Serialize damage done and resistance changes
-            var damagePrototype = prototypeManager.Index<DamageContainerPrototype>(DamageContainerId);
+            var damageContainerPrototype = _prototypeManager.Index<DamageContainerPrototype>(DamageContainerId);
 
-            SupportedClasses.Clear();
+            SupportedGroups.Clear();
             SupportedTypes.Clear();
 
-            DamageContainerId = damagePrototype.ID;
-            SupportedClasses.UnionWith(damagePrototype.SupportedClasses);
-            SupportedTypes.UnionWith(damagePrototype.SupportedTypes);
+            DamageContainerId = damageContainerPrototype.ID;
+            SupportedGroups.UnionWith(damageContainerPrototype.SupportedDamageGroups);
+            SupportedTypes.UnionWith(damageContainerPrototype.SupportedDamageTypes);
 
-            var resistancePrototype = prototypeManager.Index<ResistanceSetPrototype>(ResistanceSetId);
+            foreach (var DamageType in SupportedTypes)
+            {
+                _damageList.Add(DamageType,0);
+            }
+
+            var resistancePrototype = _prototypeManager.Index<ResistanceSetPrototype>(ResistanceSetId);
             Resistances = new ResistanceSet(resistancePrototype);
         }
 
+        public bool SupportsDamageClass(DamageGroupPrototype damageGroup)
+        {
+            return SupportedGroups.Contains(damageGroup);
+        }
+
+        public bool SupportsDamageType(DamageTypePrototype type)
+        {
+            return SupportedTypes.Contains(type);
+        }
+
         protected override void Startup()
         {
             base.Startup();
 
             ForceHealthChangedEvent();
         }
 
+        public DamageTypePrototype GetDamageType(string ID)
+        {
+            return _prototypeManager.Index<DamageTypePrototype>(ID);
+        }
+
+        public DamageGroupPrototype GetDamageGroup(string ID)
+        {
+            return _prototypeManager.Index<DamageGroupPrototype>(ID);
+        }
```

Comment:
These two methods should be removed and any usages of them replaced with the code inside each.